src/main/java/com/project/model/User.java

```diff
@@ -0,0 +1,90 @@
+package com.project.model;
+
+import com.project.model.request.UserRequest;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Entity
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+@Table(name = "user")
+public class User implements UserDetails {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @NotNull
+    @NotBlank
+    @Column(name = "email", unique = true)
+    private String email;
+
+    @NotNull
+    @NotBlank
+    @Column(name = "password")
+    private String password;
+
+    @NotBlank(message = "Vardas negali būti tuščias")
+    @Size(max = 255, message = "Vardas negali būti ilgesnis nei 255 simboliai")
+    @Column(name = "full_name")
+    private String fullName;
+
+    @OneToOne(mappedBy = "applicant")
+    private Application application;
+
+    @ElementCollection(fetch = FetchType.EAGER)
+    @Builder.Default
+    private List<String> roles = new ArrayList<>();
+
+    @Override
+    public Collection<? extends GrantedAuthority> getAuthorities() {
+        return this.roles.stream().map(SimpleGrantedAuthority::new).collect(Collectors.toList());
+    }
+
+    @Override
+    public String getUsername() {
+        return this.email;
+    }
+
+    @Override
+    public boolean isAccountNonExpired() {
+        return true;
+    }
+
+    @Override
+    public boolean isAccountNonLocked() {
+        return true;
+    }
+
+    @Override
+    public boolean isCredentialsNonExpired() {
+        return true;
+    }
+
+    @Override
+    public boolean isEnabled() {
+        return true;
+    }
+
+    public User (UserRequest user) {
+        this.fullName = user.getFullName();
+        this.email = user.getEmail();
+        this.password = user.getPassword();
+        this.roles =  new ArrayList<>(Collections.singletonList(UserRoles.USER.toString()));
```

Comment:
You can use the method `UserRoles.USER.name()` to use the text of the enum constant.

Reply:
replaced `UserRole.USER.toString()` methods to `UserRole.USER.name()